cwltool/expression.py

```diff
@@ -302,11 +303,23 @@ def interpolate(
                 return e
             leaf = json_dumps(e, sort_keys=True)
             if leaf[0] == '"':
-                leaf = leaf[1:-1]
+                leaf = json.loads(leaf)
             parts.append(leaf)
         elif scan[w[0]] == "\\":
-            e = scan[w[1] - 1]
-            parts.append(e)
+            # Backslash quoting requires a three character lookahead.
+            e = scan[w[0] : w[1] + 1]
+            if e in ("\\$(", "\\${"):
+                # Suppress start of a parameter reference, drop the
+                # backslash.
+                parts.append(e[1:])
+                w = (w[0], w[1] + 1)
+            elif e[1] == "\\":
+                # Double backslash, becomes a single backslash
+                parts.append("\\")
+            else:
+                # Some other text, add it as-is (including the
+                # backslash) and resume scanning.
+                parts.append(e[:2])
```

Comment:
Needs a check on the original `cwlVersion` of this document; shouldn't enable for less than `v1.2.0-dev4`